File: src/main/java/ipp/PrinterAttributes.java
package ipp;

import de.gmuth.ipp.client.IppPrinter;
import de.gmuth.ipp.core.IppAttribute;

public class PrinterAttributes {

  public static void main(String[] args) {
    IppPrinter ippPrinter = new IppPrinter("ipp://localhost:8632/ipp/print/laser");
    ippPrinter.logDetails();

    System.out.println("printer attributes:");
    for (IppAttribute<?> ippAttribute : ippPrinter.getAttributes().values()) {
      System.out.println(String.format("* %s = %s", ippAttribute.getName(), ippAttribute.getValues()));
    }
  }

}

File: src/main/kotlin/ipp/PrinterAttributes.kt
package ipp

import de.gmuth.ipp.client.IppPrinter
import ipp.Printers.laser

fun main() {
    val ippPrinter = IppPrinter(laser)
    ippPrinter.logDetails()

    println("printer attributes:")
    for (ippAttribute in ippPrinter.attributes.values) {
        println("* ${ippAttribute.name} = ${ippAttribute.values.joinToString(",")}")
    }
}